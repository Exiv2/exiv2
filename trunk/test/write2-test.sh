#!/bin/bash
# Test driver for write unit tests to build Exif metadata from scratch

source ./functions.source

(	cd "$testdir"

	copyTestFile        exiv2-empty.jpg
	runTest write2-test exiv2-empty.jpg

) > $results

reportTest

# That's all Folks!
##