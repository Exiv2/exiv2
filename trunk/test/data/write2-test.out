----- Some IFD0 tags
Exif.Image.Model                              0x0110 IFD0         Ascii       7 Test 1
Exif.Image.SamplesPerPixel                    0x0115 IFD0         Short       4 160 161 162 163
Exif.Image.XResolution                        0x011a IFD0         SLong       4 -2 -1 0 1
Exif.Image.YResolution                        0x011b IFD0         SRational   4 -2/3 -1/3 0/3 1/3
Exif.Image.WhitePoint                         0x013e IFD0         Undefined   4 255 254 253 252

----- One Exif tag
Exif.Image.ExifTag                            0x8769 IFD0         Long        1 26
Exif.Photo.DateTimeOriginal                   0x9003 Exif         Ascii       7 Test 2

----- Canon MakerNote tags
Exif.Image.Make                               0x010f IFD0         Ascii       6 Canon
Exif.Image.Model                              0x0110 IFD0         Ascii      20 Canon PowerShot S40
Exif.Image.ExifTag                            0x8769 IFD0         Long        1 76
Exif.Photo.MakerNote                          0x927c Exif         Undefined 118 6 0 205 171 2 0 22 0 0 0 172 0 0 0 1 0 3 0 3 0 0 0 194 0 0 0 4 0 3 0 6 0 0 0 200 0 0 0 15 0 3 0 2 0 0 0 4 0 43 0 18 0 3 0 2 0 0 0 0 0 44 0 5 0 3 0 2 0 0 0 0 0 45 0 0 0 0 0 65 32 67 97 110 111 110 32 109 97 107 101 114 110 111 116 101 32 116 97 103 0 6 0 0 0 41 0 12 0 0 0 0 0 0 0 0 0 42 0
Exif.MakerNote.Offset                         0x0001 Makernote    Long        1 94
Exif.MakerNote.ByteOrder                      0x0002 Makernote    Ascii       3 II
Exif.Canon.0xabcd                             0xabcd Makernote    Ascii      22 A Canon makernote tag
Exif.CanonCs.0x0000                           0x0000 Makernote    Short       1 6
Exif.CanonCs.Macro                            0x0001 Makernote    Short       1 0
Exif.CanonCs.Selftimer                        0x0002 Makernote    Short       1 41
Exif.CanonSi.0x0000                           0x0000 Makernote    Short       1 12
Exif.CanonSi.0x0001                           0x0001 Makernote    Short       1 0
Exif.CanonSi.ISOSpeed                         0x0002 Makernote    Short       1 0
Exif.CanonSi.MeasuredEV                       0x0003 Makernote    Short       1 0
Exif.CanonSi.TargetAperture                   0x0004 Makernote    Short       1 0
Exif.CanonSi.TargetShutterSpeed               0x0005 Makernote    Short       1 42
Exif.CanonCf.0x0000                           0x0000 Makernote    Short       1 4
Exif.CanonCf.NoiseReduction                   0x0001 Makernote    Short       1 43
Exif.CanonPi.0x0000                           0x0000 Makernote    Short       1 0
Exif.CanonPi.0x0001                           0x0001 Makernote    Short       1 44
Exif.CanonPa.0x0000                           0x0000 Makernote    Short       1 0
Exif.CanonPa.0x0001                           0x0001 Makernote    Short       1 45

----- Non-intrusive writing of special Canon MakerNote tags
Exif.Image.Make                               0x010f IFD0         Ascii       6 Canon
Exif.Image.Model                              0x0110 IFD0         Ascii      20 Canon PowerShot S40
Exif.Image.ExifTag                            0x8769 IFD0         Long        1 76
Exif.Photo.MakerNote                          0x927c Exif         Undefined 118 6 0 205 171 2 0 22 0 0 0 172 0 0 0 1 0 3 0 3 0 0 0 194 0 0 0 4 0 3 0 6 0 0 0 200 0 0 0 15 0 3 0 2 0 0 0 4 0 43 0 18 0 3 0 2 0 0 0 0 0 44 0 5 0 3 0 2 0 0 0 0 0 45 0 0 0 0 0 65 32 67 97 110 111 110 32 109 97 107 101 114 110 111 116 101 32 116 97 103 0 6 0 88 0 41 0 12 0 0 0 0 0 0 0 99 0 42 0
Exif.MakerNote.Offset                         0x0001 Makernote    Long        1 94
Exif.MakerNote.ByteOrder                      0x0002 Makernote    Ascii       3 II
Exif.Canon.0xabcd                             0xabcd Makernote    Ascii      22 A Canon makernote tag
Exif.CanonCs.0x0000                           0x0000 Makernote    Short       1 6
Exif.CanonCs.Macro                            0x0001 Makernote    Short       1 88
Exif.CanonCs.Selftimer                        0x0002 Makernote    Short       1 41
Exif.CanonSi.0x0000                           0x0000 Makernote    Short       1 12
Exif.CanonSi.0x0001                           0x0001 Makernote    Short       1 0
Exif.CanonSi.ISOSpeed                         0x0002 Makernote    Short       1 0
Exif.CanonSi.MeasuredEV                       0x0003 Makernote    Short       1 0
Exif.CanonSi.TargetAperture                   0x0004 Makernote    Short       1 99
Exif.CanonSi.TargetShutterSpeed               0x0005 Makernote    Short       1 42
Exif.CanonCf.0x0000                           0x0000 Makernote    Short       1 4
Exif.CanonCf.NoiseReduction                   0x0001 Makernote    Short       1 43
Exif.CanonPi.0x0000                           0x0000 Makernote    Short       1 0
Exif.CanonPi.0x0001                           0x0001 Makernote    Short       1 44
Exif.CanonPa.0x0000                           0x0000 Makernote    Short       1 0
Exif.CanonPa.0x0001                           0x0001 Makernote    Short       1 45

----- One Fujifilm MakerNote tag
Exif.Image.Make                               0x010f IFD0         Ascii       9 FUJIFILM
Exif.Image.Model                              0x0110 IFD0         Ascii      13 FinePixS2Pro
Exif.Image.ExifTag                            0x8769 IFD0         Long        1 74
Exif.Photo.MakerNote                          0x927c Exif         Undefined  54 70 85 74 73 70 73 76 77 12 0 0 0 1 0 0 16 2 0 23 0 0 0 30 0 0 0 0 0 0 0 65 32 70 117 106 105 102 105 108 109 32 81 85 65 76 73 84 89 32 116 97 103 0 0
Exif.MakerNote.Offset                         0x0001 Makernote    Long        1 92
Exif.MakerNote.ByteOrder                      0x0002 Makernote    Ascii       3 II
Exif.Fujifilm.Quality                         0x1000 Makernote    Ascii      23 A Fujifilm QUALITY tag

----- One Sigma/Foveon MakerNote tag
Exif.Image.Make                               0x010f IFD0         Ascii       6 SIGMA
Exif.Image.Model                              0x0110 IFD0         Ascii      11 SIGMA SD10
Exif.Image.ExifTag                            0x8769 IFD0         Long        1 68
Exif.Photo.MakerNote                          0x927c Exif         Undefined  46 83 73 71 77 65 0 0 0 1 0 1 0 24 0 2 0 17 0 0 0 114 0 0 0 0 0 0 0 83 111 102 116 119 97 114 101 63 32 69 120 105 118 50 33 0 0
Exif.MakerNote.Offset                         0x0001 Makernote    Long        1 86
Exif.MakerNote.ByteOrder                      0x0002 Makernote    Ascii       3 II
Exif.Sigma.Software                           0x0018 Makernote    Ascii      17 Software? Exiv2!

----- One Nikon1 MakerNote tag
Exif.Image.Make                               0x010f IFD0         Ascii       6 NIKON
Exif.Image.Model                              0x0110 IFD0         Ascii       5 E990
Exif.Image.ExifTag                            0x8769 IFD0         Long        1 62
Exif.Photo.MakerNote                          0x927c Exif         Undefined  44 1 0 128 0 2 0 25 0 0 0 98 0 0 0 0 0 0 0 73 109 97 103 101 65 100 106 117 115 116 109 101 110 116 32 98 121 32 69 120 105 118 50 0 0
Exif.MakerNote.Offset                         0x0001 Makernote    Long        1 80
Exif.MakerNote.ByteOrder                      0x0002 Makernote    Ascii       3 II
Exif.Nikon1.ImageAdjustment                   0x0080 Makernote    Ascii      25 ImageAdjustment by Exiv2

----- One Nikon2 MakerNote tag
Exif.Image.Make                               0x010f IFD0         Ascii       6 NIKON
Exif.Image.Model                              0x0110 IFD0         Ascii       5 E950
Exif.Image.ExifTag                            0x8769 IFD0         Long        1 62
Exif.Photo.MakerNote                          0x927c Exif         Undefined  48 78 105 107 111 110 0 1 0 1 0 255 255 2 0 22 0 0 0 106 0 0 0 0 0 0 0 65 110 32 111 98 115 99 117 114 101 32 78 105 107 111 110 50 32 116 97 103 0
Exif.MakerNote.Offset                         0x0001 Makernote    Long        1 80
Exif.MakerNote.ByteOrder                      0x0002 Makernote    Ascii       3 II
Exif.Nikon2.0xffff                            0xffff Makernote    Ascii      22 An obscure Nikon2 tag

----- One Nikon3 MakerNote tag
Exif.Image.Make                               0x010f IFD0         Ascii      18 NIKON CORPORATION
Exif.Image.Model                              0x0110 IFD0         Ascii      10 NIKON D70
Exif.Image.ExifTag                            0x8769 IFD0         Long        1 78
Exif.Photo.MakerNote                          0x927c Exif         Undefined  64 78 105 107 111 110 0 2 16 0 0 73 73 42 0 8 0 0 0 1 0 4 0 2 0 28 0 0 0 26 0 0 0 0 0 0 0 65 32 98 111 114 105 110 103 32 78 105 107 111 110 51 32 81 117 97 108 105 116 121 32 116 97 103 0
Exif.MakerNote.Offset                         0x0001 Makernote    Long        1 96
Exif.MakerNote.ByteOrder                      0x0002 Makernote    Ascii       3 II
Exif.Nikon3.Quality                           0x0004 Makernote    Ascii      28 A boring Nikon3 Quality tag

----- One Olympus MakerNote tag
Exif.Image.Make                               0x010f IFD0         Ascii      20 OLYMPUS CORPORATION
Exif.Image.Model                              0x0110 IFD0         Ascii       8 C8080WZ
Exif.Image.ExifTag                            0x8769 IFD0         Long        1 78
Exif.Photo.MakerNote                          0x927c Exif         Undefined  26 79 76 89 77 80 0 1 0 1 0 1 2 3 0 1 0 0 0 1 0 0 0 0 0 0 0
Exif.MakerNote.Offset                         0x0001 Makernote    Long        1 96
Exif.MakerNote.ByteOrder                      0x0002 Makernote    Ascii       3 II
Exif.Olympus.Quality                          0x0201 Makernote    Short       1 1

----- One Panasonic MakerNote tag
Exif.Image.Make                               0x010f IFD0         Ascii      10 Panasonic
Exif.Image.Model                              0x0110 IFD0         Ascii       8 DMC-FZ5
Exif.Image.ExifTag                            0x8769 IFD0         Long        1 68
Exif.Photo.MakerNote                          0x927c Exif         Undefined  26 80 97 110 97 115 111 110 105 99 0 0 0 1 0 1 0 3 0 1 0 0 0 1 0 0 0
Exif.MakerNote.Offset                         0x0001 Makernote    Long        1 86
Exif.MakerNote.ByteOrder                      0x0002 Makernote    Ascii       3 II
Exif.Panasonic.Quality                        0x0001 Makernote    Short       1 1

----- One Sony1 MakerNote tag
Exif.Image.Make                               0x010f IFD0         Ascii       5 SONY
Exif.Image.Model                              0x0110 IFD0         Ascii       7 DSC-W7
Exif.Image.ExifTag                            0x8769 IFD0         Long        1 64
Exif.Photo.MakerNote                          0x927c Exif         Undefined  32 83 79 78 89 32 68 83 67 32 0 0 0 1 0 0 32 7 0 6 0 0 0 108 0 0 0 0 1 2 3 4 5
Exif.MakerNote.Offset                         0x0001 Makernote    Long        1 82
Exif.MakerNote.ByteOrder                      0x0002 Makernote    Ascii       3 II
Exif.Sony1.0x2000                             0x2000 Makernote    Undefined   6 0 1 2 3 4 5

----- Minolta MakerNote tags
Exif.Image.Make                               0x010f IFD0         Ascii       8 Minolta
Exif.Image.Model                              0x0110 IFD0         Ascii      23 A fancy Minolta camera
Exif.Image.ExifTag                            0x8769 IFD0         Long        1 82
Exif.Photo.MakerNote                          0x927c Exif         Undefined 346 4 0 1 1 4 0 1 0 0 0 1 0 0 0 3 0 7 0 16 0 0 0 154 0 0 0 20 1 7 0 148 0 0 0 170 0 0 0 4 0 7 0 128 0 0 0 62 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 3 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 255 255 0 0 0 0 0 0 0 0 0 4 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 255 254 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 255 253
Exif.MakerNote.Offset                         0x0001 Makernote    Long        1 100
Exif.MakerNote.ByteOrder                      0x0002 Makernote    Ascii       3 II
Exif.Minolta.ColorMode                        0x0101 Makernote    Long        1 1
Exif.MinoltaCsNew.0x0000                      0x0000 Makernote    Long        1 0
Exif.MinoltaCsNew.ExposureMode                0x0001 Makernote    Long        1 0
Exif.MinoltaCsNew.FlashMode                   0x0002 Makernote    Long        1 0
Exif.MinoltaCsNew.WhiteBalance                0x0003 Makernote    Long        1 2
Exif.MinoltaCs5D.0x0000                       0x0000 Makernote    Short       1 0
Exif.MinoltaCs5D.0x0001                       0x0001 Makernote    Short       1 0
Exif.MinoltaCs5D.0x0002                       0x0002 Makernote    Short       1 0
Exif.MinoltaCs5D.0x0003                       0x0003 Makernote    Short       1 0
Exif.MinoltaCs5D.0x0004                       0x0004 Makernote    Short       1 0
Exif.MinoltaCs5D.0x0005                       0x0005 Makernote    Short       1 0
Exif.MinoltaCs5D.0x0006                       0x0006 Makernote    Short       1 0
Exif.MinoltaCs5D.0x0007                       0x0007 Makernote    Short       1 0
Exif.MinoltaCs5D.0x0008                       0x0008 Makernote    Short       1 0
Exif.MinoltaCs5D.0x0009                       0x0009 Makernote    Short       1 0
Exif.MinoltaCs5D.ExposureMode                 0x000a Makernote    Short       1 0
Exif.MinoltaCs5D.0x000b                       0x000b Makernote    Short       1 0
Exif.MinoltaCs5D.ImageSize                    0x000c Makernote    Short       1 0
Exif.MinoltaCs5D.Quality                      0x000d Makernote    Short       1 0
Exif.MinoltaCs5D.WhiteBalance                 0x000e Makernote    Short       1 3
Exif.MinoltaCs5D.0x000f                       0x000f Makernote    Short       1 0
Exif.MinoltaCs5D.0x0010                       0x0010 Makernote    Short       1 0
Exif.MinoltaCs5D.0x0011                       0x0011 Makernote    Short       1 0
Exif.MinoltaCs5D.0x0012                       0x0012 Makernote    Short       1 0
Exif.MinoltaCs5D.0x0013                       0x0013 Makernote    Short       1 0
Exif.MinoltaCs5D.0x0014                       0x0014 Makernote    Short       1 0
Exif.MinoltaCs5D.0x0015                       0x0015 Makernote    Short       1 0
Exif.MinoltaCs5D.0x0016                       0x0016 Makernote    Short       1 0
Exif.MinoltaCs5D.0x0017                       0x0017 Makernote    Short       1 0
Exif.MinoltaCs5D.0x0018                       0x0018 Makernote    Short       1 0
Exif.MinoltaCs5D.0x0019                       0x0019 Makernote    Short       1 0
Exif.MinoltaCs5D.FocusPosition                0x001a Makernote    Short       1 0
Exif.MinoltaCs5D.FocusArea                    0x001b Makernote    Short       1 0
Exif.MinoltaCs5D.0x001c                       0x001c Makernote    Short       1 0
Exif.MinoltaCs5D.0x001d                       0x001d Makernote    Short       1 0
Exif.MinoltaCs5D.0x001e                       0x001e Makernote    Short       1 0
Exif.MinoltaCs5D.FlashFired                   0x001f Makernote    Short       1 0
Exif.MinoltaCs5D.0x0020                       0x0020 Makernote    Short       1 0
Exif.MinoltaCs5D.0x0021                       0x0021 Makernote    Short       1 0
Exif.MinoltaCs5D.0x0022                       0x0022 Makernote    Short       1 0
Exif.MinoltaCs5D.0x0023                       0x0023 Makernote    Short       1 0
Exif.MinoltaCs5D.0x0024                       0x0024 Makernote    Short       1 0
Exif.MinoltaCs5D.MeteringMode                 0x0025 Makernote    Short       1 0
Exif.MinoltaCs5D.ISOSpeed                     0x0026 Makernote    Short       1 0
Exif.MinoltaCs5D.0x0027                       0x0027 Makernote    Short       1 0
Exif.MinoltaCs5D.0x0028                       0x0028 Makernote    Short       1 0
Exif.MinoltaCs5D.0x0029                       0x0029 Makernote    Short       1 0
Exif.MinoltaCs5D.0x002a                       0x002a Makernote    Short       1 0
Exif.MinoltaCs5D.0x002b                       0x002b Makernote    Short       1 0
Exif.MinoltaCs5D.0x002c                       0x002c Makernote    Short       1 0
Exif.MinoltaCs5D.0x002d                       0x002d Makernote    Short       1 0
Exif.MinoltaCs5D.0x002e                       0x002e Makernote    Short       1 0
Exif.MinoltaCs5D.ColorSpace                   0x002f Makernote    Short       1 0
Exif.MinoltaCs5D.Sharpness                    0x0030 Makernote    Short       1 0
Exif.MinoltaCs5D.Contrast                     0x0031 Makernote    Short       1 0
Exif.MinoltaCs5D.Saturation                   0x0032 Makernote    Short       1 0
Exif.MinoltaCs5D.0x0033                       0x0033 Makernote    Short       1 0
Exif.MinoltaCs5D.0x0034                       0x0034 Makernote    Short       1 0
Exif.MinoltaCs5D.ExposureTime                 0x0035 Makernote    Short       1 0
Exif.MinoltaCs5D.FNumber                      0x0036 Makernote    Short       1 0
Exif.MinoltaCs5D.FreeMemoryCardImages         0x0037 Makernote    Short       1 0
Exif.MinoltaCs5D.ExposureRevision             0x0038 Makernote    Short       1 0
Exif.MinoltaCs5D.0x0039                       0x0039 Makernote    Short       1 0
Exif.MinoltaCs5D.0x003a                       0x003a Makernote    Short       1 0
Exif.MinoltaCs5D.0x003b                       0x003b Makernote    Short       1 0
Exif.MinoltaCs5D.0x003c                       0x003c Makernote    Short       1 0
Exif.MinoltaCs5D.0x003d                       0x003d Makernote    Short       1 0
Exif.MinoltaCs5D.0x003e                       0x003e Makernote    Short       1 0
Exif.MinoltaCs5D.0x003f                       0x003f Makernote    Short       1 0
Exif.MinoltaCs5D.0x0040                       0x0040 Makernote    Short       1 0
Exif.MinoltaCs5D.0x0041                       0x0041 Makernote    Short       1 0
Exif.MinoltaCs5D.0x0042                       0x0042 Makernote    Short       1 0
Exif.MinoltaCs5D.0x0043                       0x0043 Makernote    Short       1 0
Exif.MinoltaCs5D.0x0044                       0x0044 Makernote    Short       1 0
Exif.MinoltaCs5D.0x0045                       0x0045 Makernote    Short       1 0
Exif.MinoltaCs5D.0x0046                       0x0046 Makernote    Short       1 0
Exif.MinoltaCs5D.0x0047                       0x0047 Makernote    Short       1 0
Exif.MinoltaCs5D.FocusMode                    0x0048 Makernote    Short       1 0
Exif.MinoltaCs5D.ColorTemperature             0x0049 Makernote    SShort      1 -1
Exif.MinoltaCs7D.ExposureMode                 0x0000 Makernote    Short       1 0
Exif.MinoltaCs7D.0x0001                       0x0001 Makernote    Short       1 0
Exif.MinoltaCs7D.ImageSize                    0x0002 Makernote    Short       1 0
Exif.MinoltaCs7D.Quality                      0x0003 Makernote    Short       1 0
Exif.MinoltaCs7D.WhiteBalance                 0x0004 Makernote    Short       1 4
Exif.MinoltaCs7D.0x0005                       0x0005 Makernote    Short       1 0
Exif.MinoltaCs7D.0x0006                       0x0006 Makernote    Short       1 0
Exif.MinoltaCs7D.0x0007                       0x0007 Makernote    Short       1 0
Exif.MinoltaCs7D.0x0008                       0x0008 Makernote    Short       1 0
Exif.MinoltaCs7D.0x0009                       0x0009 Makernote    Short       1 0
Exif.MinoltaCs7D.0x000a                       0x000a Makernote    Short       1 0
Exif.MinoltaCs7D.0x000b                       0x000b Makernote    Short       1 0
Exif.MinoltaCs7D.0x000c                       0x000c Makernote    Short       1 0
Exif.MinoltaCs7D.0x000d                       0x000d Makernote    Short       1 0
Exif.MinoltaCs7D.FocusMode                    0x000e Makernote    Short       1 0
Exif.MinoltaCs7D.0x000f                       0x000f Makernote    Short       1 0
Exif.MinoltaCs7D.AFPoints                     0x0010 Makernote    Short       1 0
Exif.MinoltaCs7D.0x0011                       0x0011 Makernote    Short       1 0
Exif.MinoltaCs7D.0x0012                       0x0012 Makernote    Short       1 0
Exif.MinoltaCs7D.0x0013                       0x0013 Makernote    Short       1 0
Exif.MinoltaCs7D.0x0014                       0x0014 Makernote    Short       1 0
Exif.MinoltaCs7D.FlashFired                   0x0015 Makernote    Short       1 0
Exif.MinoltaCs7D.FlashMode                    0x0016 Makernote    Short       1 0
Exif.MinoltaCs7D.0x0017                       0x0017 Makernote    Short       1 0
Exif.MinoltaCs7D.0x0018                       0x0018 Makernote    Short       1 0
Exif.MinoltaCs7D.0x0019                       0x0019 Makernote    Short       1 0
Exif.MinoltaCs7D.0x001a                       0x001a Makernote    Short       1 0
Exif.MinoltaCs7D.0x001b                       0x001b Makernote    Short       1 0
Exif.MinoltaCs7D.ISOSpeed                     0x001c Makernote    Short       1 0
Exif.MinoltaCs7D.0x001d                       0x001d Makernote    Short       1 0
Exif.MinoltaCs7D.ExposureCompensation         0x001e Makernote    SShort      1 -2
Exif.MinoltaCs7D.0x001f                       0x001f Makernote    Short       1 0
Exif.MinoltaCs7D.0x0020                       0x0020 Makernote    Short       1 0
Exif.MinoltaCs7D.0x0021                       0x0021 Makernote    Short       1 0
Exif.MinoltaCs7D.0x0022                       0x0022 Makernote    Short       1 0
Exif.MinoltaCs7D.0x0023                       0x0023 Makernote    Short       1 0
Exif.MinoltaCs7D.0x0024                       0x0024 Makernote    Short       1 0
Exif.MinoltaCs7D.ColorSpace                   0x0025 Makernote    Short       1 0
Exif.MinoltaCs7D.Sharpness                    0x0026 Makernote    Short       1 0
Exif.MinoltaCs7D.Contrast                     0x0027 Makernote    Short       1 0
Exif.MinoltaCs7D.Saturation                   0x0028 Makernote    Short       1 0
Exif.MinoltaCs7D.0x0029                       0x0029 Makernote    Short       1 0
Exif.MinoltaCs7D.0x002a                       0x002a Makernote    Short       1 0
Exif.MinoltaCs7D.0x002b                       0x002b Makernote    Short       1 0
Exif.MinoltaCs7D.0x002c                       0x002c Makernote    Short       1 0
Exif.MinoltaCs7D.FreeMemoryCardImages         0x002d Makernote    Short       1 0
Exif.MinoltaCs7D.0x002e                       0x002e Makernote    Short       1 0
Exif.MinoltaCs7D.0x002f                       0x002f Makernote    Short       1 0
Exif.MinoltaCs7D.0x0030                       0x0030 Makernote    Short       1 0
Exif.MinoltaCs7D.0x0031                       0x0031 Makernote    Short       1 0
Exif.MinoltaCs7D.0x0032                       0x0032 Makernote    Short       1 0
Exif.MinoltaCs7D.0x0033                       0x0033 Makernote    Short       1 0
Exif.MinoltaCs7D.0x0034                       0x0034 Makernote    Short       1 0
Exif.MinoltaCs7D.0x0035                       0x0035 Makernote    Short       1 0
Exif.MinoltaCs7D.0x0036                       0x0036 Makernote    Short       1 0
Exif.MinoltaCs7D.0x0037                       0x0037 Makernote    Short       1 0
Exif.MinoltaCs7D.0x0038                       0x0038 Makernote    Short       1 0
Exif.MinoltaCs7D.0x0039                       0x0039 Makernote    Short       1 0
Exif.MinoltaCs7D.0x003a                       0x003a Makernote    Short       1 0
Exif.MinoltaCs7D.0x003b                       0x003b Makernote    Short       1 0
Exif.MinoltaCs7D.0x003c                       0x003c Makernote    Short       1 0
Exif.MinoltaCs7D.0x003d                       0x003d Makernote    Short       1 0
Exif.MinoltaCs7D.0x003e                       0x003e Makernote    Short       1 0
Exif.MinoltaCs7D.ColorTemperature             0x003f Makernote    SShort      1 -3

----- One IOP tag
Exif.Image.ExifTag                            0x8769 IFD0         Long        1 26
Exif.Photo.InteroperabilityTag                0xa005 Exif         Long        1 44
Exif.Iop.InteroperabilityIndex                0x0001 Iop          Ascii       7 Test 3

----- One GPS tag
Exif.Image.GPSTag                             0x8825 IFD0         Long        1 26
Exif.GPSInfo.GPSVersionID                     0x0000 GPSInfo      Byte        2 19 20

----- One IFD1 tag
Exif.Thumbnail.Artist                         0x013b IFD1         Ascii       7 Test 5

----- One IOP and one IFD1 tag
Exif.Image.ExifTag                            0x8769 IFD0         Long        1 26
Exif.Photo.InteroperabilityTag                0xa005 Exif         Long        1 44
Exif.Iop.InteroperabilityIndex                0x0001 Iop          Ascii      15 Test 6 Iop tag
Exif.Thumbnail.Artist                         0x013b IFD1         Ascii      16 Test 6 Ifd1 tag

----- One IFD0 and one IFD1 tag
Exif.Image.SamplesPerPixel                    0x0115 IFD0         Short       4 160 161 162 163
Exif.Thumbnail.Artist                         0x013b IFD1         Ascii       7 Test 7
